osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
+ * 3) Final Strain is the summation of global and column strain.
+ */
+
 namespace osu.Game.Rulesets.Mania.Difficulty.Skills
 {
     public class Strain : StrainDecaySkill
     {
-        private const double individual_decay_base = 0.125;
-        private const double overall_decay_base = 0.30;
+        /// <summary> Individual Strain Decay Exponent Base. Used in <see cref="applyDecay"/> </summary>
+        private const double decay_base = 0.125;
+
+        /// <summary> Global Strain Decay Exponent Base. Used in <see cref="applyDecay"/> </summary>
+        private const double global_decay_base = 0.30;
+
+        /// <summary> Center of our endOnBodyBias sigmoid function. </summary>
         private const double release_threshold = 24;
 
         protected override double SkillMultiplier => 1;
         protected override double StrainDecayBase => 1;
 
-        private readonly double[] startTimes;
-        private readonly double[] endTimes;
-        private readonly double[] individualStrains;
+        /// <summary> Previous notes' start times. Indices correspond to columns </summary>
+        private readonly double[] prevStartTimes;
+
+        /// <summary> Previous notes' end times. Indices correspond to columns </summary>
+        private readonly double[] prevEndTimes;
+
+        /// <summary> Per-Column Strains. Indices correspond to columns </summary>
+        private readonly double[] columnStrains;
+
+        /// <summary> Previous Strain processed. </summary>
+        private double prevColumnStrain;
 
-        private double individualStrain;
-        private double overallStrain;
+        /// <summary> Global Strain: The "strain" of all fingers. </summary>
+        private double globalStrain;
 
         public Strain(Mod[] mods, int totalColumns)
             : base(mods)
         {
-            startTimes = new double[totalColumns];
-            endTimes = new double[totalColumns];
-            individualStrains = new double[totalColumns];
-            overallStrain = 1;
+            prevStartTimes = new double[totalColumns];
+            prevEndTimes = new double[totalColumns];
+            columnStrains = new double[totalColumns];
+            globalStrain = 1;
         }
 
+        /// <summary>
+        /// Calculates the strain value of a <see cref="DifficultyHitObject"/>. This value is affected by previously processed objects.
+        ///
+        /// The function documentation heavily references diagrams the top of the file.
+        ///
+        /// <param name="current">Current Hit Object to evaluate strain</param>
+        /// <remarks>
+        /// 1) The first hitObject is not considered in the calculation.
+        /// <see cref="ManiaDifficultyCalculator.CreateDifficultyHitObjects"/>
+        /// 2) Strain is binned/bucketed by sections by Section Length, with a Max aggregation
+        /// <see cref="StrainSkill.SectionLength"/>
+        /// <see cref="StrainSkill.Process"/>
```

Comment:
I mean that is how all strain skills work, so I'm not sure whether it is worth to restate this here?